### ISSUE TITLE
List of variables for Docker

### ISSUE THREAD

**Issue by modem7** [almarklein/timetagger#294]
Heya, 

I looked at the docs, but couldn't see anything regarding this. 

Is there a list of all the variables that we can use within a Docker deployment for the app? 

I could only find:

```
TIMETAGGER_BIND
TIMETAGGER_DATADIR
TIMETAGGER_LOG_LEVEL
TIMETAGGER_CREDENTIALS
```

Given the amount of additional settings that are available, especially with the latest release (thank you!), I'm wondering if there is an easier way to implement via the variables. 

Thanks!

**Comment by almarklein** [comment]
All server configuration options are listed [here](https://timetagger.readthedocs.io/en/latest/libapi/#configuration).

Settings such as the time format are user settings that can be set (by each user) using the UI (via the settings dialog). Technically they could also be set via the [web api](https://timetagger.readthedocs.io/en/latest/webapi/).

I think what you're hinting at may be the option to set default values for these user-settings via server config variables? This is not currently supported.

**Comment by almarklein** [comment]
I think it could be. Could you explain a bit more about the use-case? Currently I don't see that much of a benefit over just using the settings dialog 😄

**Comment by drajabr** [comment]
@almarklein @modem7 found proxy vars easily, apparently, I'm the noob who couldn't figure out how to use them properly. 
`TIMETAGGER_PROXY_AUTH_ENABLED
TIMETAGGER_PROXY_AUTH_TRUSTED
TIMETAGGER_PROXY_AUTH_HEADER`

**Comment by modem7** [comment]
> I think it could be. Could you explain a bit more about the use-case? Currently I don't see that much of a benefit over just using the settings dialog 😄 

So sorry that this ticket was forgotten about!

So, from a docker deployment perspective, anything that can be set prior to deployment is beneficial, especially if sharing configs with others/troubleshooting etc.

Whilst not strictly necessary, it is useful, rather than making sure UI elements and options are filled in in each individual deployment when an env var would solve this for all users in a much more streamlined way.

**Comment by almarklein** [comment]
Right. Well, the defaults are currently hardcoded. Replacing them with a call to `os.getenv()` seems reasonable!
